File: FRONTEND/remove_comments.py

import os
import re

def remove_comments(content):
    # Regex to match strings or comments
    pattern = r'("(\\.|[^"\\])*"|\'(\\.|[^\'\\])*\'|`(\\.|[^`\\])*`)|(\/\/[^\r\n]*)|(\/\*[\s\S]*?\*\/)'
    
    def replacer(match):
        # If it's a string (group 1), return it as is
        if match.group(1):
            return match.group(1)
        # Otherwise it's a comment, return empty string
        return ""
    
    return re.sub(pattern, replacer, content)

def process_directory(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.js', '.jsx')):
                file_path = os.path.join(root, file)
                try:
                    with open(file_path, 'r', encoding='utf-8') as f:
                        content = f.read()
                    
                    new_content = remove_comments(content)
                    
                    # Basic cleanup of empty lines left by full-line comments
                    # new_content = re.sub(r'^\s*$\n', '', new_content, flags=re.MULTILINE)

                    if content != new_content:
                        with open(file_path, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        print(f"Cleaned: {file}")
                except Exception as e:
                    print(f"Error processing {file}: {e}")

if __name__ == "__main__":
    print("Starting comment removal...")
    process_directory("src")
    print("Comment removal complete.")

File: BACKEND/FRONTEND/remove_comments.js

const fs = require('fs');
const path = require('path');

function removeComments(content) {
    // Regex to match strings or comments
    const pattern = /("(\\.|[^"\\])*"|'(\\.|[^'\\])*'|`(\\.|[^`\\])*`)|(\/\/[^\r\n]*)|(\/\*[\s\S]*?\*\/)/g;

    return content.replace(pattern, (match, string) => {
        // If it's a string (captured in the first group), return it as is
        if (string) {
            return string;
        }
        // Otherwise it's a comment, return empty string
        return "";
    });
}

function processDirectory(directory) {
    const files = fs.readdirSync(directory);

    files.forEach(file => {
        const fullPath = path.join(directory, file);
        const stat = fs.statSync(fullPath);

        if (stat.isDirectory()) {
            processDirectory(fullPath);
        } else if (file.endsWith('.js') || file.endsWith('.jsx')) {
            try {
                const content = fs.readFileSync(fullPath, 'utf8');
                const newContent = removeComments(content);

                if (content !== newContent) {
                    fs.writeFileSync(fullPath, newContent, 'utf8');
                    console.log(`Cleaned: ${file}`);
                }
            } catch (err) {
                console.error(`Error processing ${file}: ${err.message}`);
            }
        }
    });
}

console.log("Starting comment removal...");
processDirectory("src");
console.log("Comment removal complete.");
